[backend] Microsoft Defender collector not working (#1686)

antoinemzs opened the pull request:
<!--
Thank you very much for your pull request to the OpenBAS project! We as a community driven project depend on support and contributions like this!

Thus already a BIG THANK YOU upfront to you for choosing to help with your PR.
-->

### Proposed changes

* Rework the microsoft defender collector to use a bespoke query to fetch alert data tied to a process tree.
* Only match machine name and the presence of the specific parent process tagged with the inject uuid

### Related issues

* Contributes to https://github.com/OpenBAS-Platform/openbas/issues/1686

### Checklist

<!--
Please submit the source code in a way, where you could honestly say `This code is finished`.
If you feel that there are possibilities for improving the code quality, please do so.
By doing this, you are actively helping us to improve the quality of the entire OpenBAS project.
-->

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality using different use cases
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

<!-- For completed items, change [ ] to [x]. -->

### Further comments

<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->

antoinemzs committed:
change query to threat hunting api

Signed-off-by: Antoine MAZEAS <antoine.mazeas@filigran.io>

antoinemzs committed:
working query for openbas implant

Signed-off-by: Antoine MAZEAS <antoine.mazeas@filigran.io>

antoinemzs committed:
run isort

Signed-off-by: Antoine MAZEAS <antoine.mazeas@filigran.io>

antoinemzs committed:
remove unused import

Signed-off-by: Antoine MAZEAS <antoine.mazeas@filigran.io>

RomuDeuxfois commented on microsoft-defender/src/openbas_microsoft_defender.py:
```diff
@@ -226,17 +238,22 @@ async def _process_alerts(self, graph_client):
         )
+
+        if not any(expectations):
+            self.helper.collector_logger.info("No work: skipping iteration.")
```
**suggestion:**
self.helper.collector_logger.info("No expectations found: skipping iteration.")
It's more readable ?

antoinemzs committed:
handle executors appearing as child or parent proc

Some injects may be only detected but not prevented, resulting in an
"evidence" that may show the signature process name as a child
"process_name" or parent process.

We cannot yet build a logical collection of AND/OR required or optional
signatures on expectations, so executors may define a "process_name"
meaning any process being a child or a parent.

Signed-off-by: Antoine MAZEAS <antoine.mazeas@filigran.io>

antoinemzs committed:
reinstate fuzzy matcher for parent process name

Signed-off-by: Antoine MAZEAS <antoine.mazeas@filigran.io>

antoinemzs committed:
correct filter child/parent procs w/ obas-implant

Signed-off-by: Antoine MAZEAS <antoine.mazeas@filigran.io>
